tests/dragonfly/acl_family_test.py

```diff
@@ -635,3 +645,83 @@ async def test_auth_resp3_bug(df_factory):
     }
 
     await client.close()
+
+
+@pytest.mark.asyncio
+async def test_acl_pub_sub_auth(df_factory):
+    df = df_factory.create()
+    df.start()
+    client = df.client()
+    await client.execute_command("ACL SETUSER kostas on >tmp +subscribe +psubscribe &f*o &bar")
+    assert await client.execute_command("AUTH kostas tmp") == "OK"
+
+    res = await client.execute_command("SUBSCRIBE bar")
+    assert res == ["subscribe", "bar", 1]
+
+    res = await client.execute_command("SUBSCRIBE foo")
+    assert res == ["subscribe", "foo", 2]
+
+    with pytest.raises(redis.exceptions.NoPermissionError):
+        res = await client.execute_command("SUBSCRIBE my_channel")
+
+    # PSUBSCRIBE only matches pure literals, no asterisks
+    with pytest.raises(redis.exceptions.NoPermissionError):
+        res = await client.execute_command("PSUBSCRIBE foo")
+
+    # my_channel is not in our list so the command should fail
+    with pytest.raises(redis.exceptions.NoPermissionError):
+        res = await client.execute_command("PSUBSCRIBE bar my_channel")
+
+    res = await client.execute_command("PSUBSCRIBE bar")
+    assert res == ["psubscribe", "bar", 3]
+
+
+@pytest.mark.asyncio
+async def test_acl_revoke_pub_sub_while_subscribed(df_factory):
```

Comment:
@dranikpg I added this test to be sure that we are on part with Valkey. It's the exact same behaviour:

1. We only evict connections when `resetchannels` is passed
2. If we remove `subscribe/unsubscribe` while a client is already `subcribed` to a channel it has `no effect` meaning that the client will still be able to `receive/consume from the channel unaffected` and won't be able to `unsbuscribe` :rofl: (unless it explicitly kills the client). That second part is like blackmailing, "I am forcing you to consume the channel forever") 

P.s. I run the test on Valkey as well.

Reply:
It this behaviour is intended I'm okay with it 🙂